web/docs/auth/entities.md

```diff
@@ -0,0 +1,508 @@
+
+```tsx title="src/client/MainPage.tsx"
+import { getFirstProviderUserId } from '@wasp/auth/user'
+import { User } from '@wasp/auth/types'
+
+const MainPage = ({ user }: { user: User }) => {
+  const userId = getFirstProviderUserId(user)
+  // ...
+}
+```
+
+```ts title=src/server/tasks.ts
+import { getFirstProviderUserId } from '@wasp/auth/user.js'
+
+export const createTask: CreateTask<...>  = async (args, context) => {
+  const userId = getFirstProviderUserId(context.user)
+  // ...
+}
+```
+
+</TabItem>
+</Tabs>
+
+### `findUserIdentity`
+
+You can find a specific auth identity by using the `findUserIdentity` helper function. This function takes a `user` and a `providerName` and returns the first auth identity that matches the `providerName` or `null` if no auth identity matches the `providerName`.
+
+This can be useful if you want to check if the user has a specific auth identity. For example, you might want to check if the user has an email-based auth identity.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx title="src/client/MainPage.jsx"
+import { findUserIdentity } from '@wasp/auth/user'
+
+const MainPage = ({ user }) => {
+  const emailIdentity = findUserIdentity(user, 'email')
+  const googleIdentity = findUserIdentity(user, 'google')
+  if (emailIdentity) {
+    // ...
+  } else if (googleIdentity) {
+    // ...
+  }
+  // ...
+}
+```
+
+```js title=src/server/tasks.js
+import { findUserIdentity } from '@wasp/auth/user.js'
+
+export const createTask = async (args, context) => {
+  const emailIdentity = findUserIdentity(context.user, 'email')
+  const googleIdentity = findUserIdentity(context.user, 'google')
+  if (emailIdentity) {
+    // ...
+  } else if (googleIdentity) {
+    // ...
+  }
+  // ...
+}
+```
+
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="src/client/MainPage.tsx"
+import { findUserIdentity } from '@wasp/auth/user'
+import { User } from '@wasp/auth/types'
+
+const MainPage = ({ user }: { user: User }) => {
+  const emailIdentity = findUserIdentity(user, 'email')
+  const googleIdentity = findUserIdentity(user, 'google')
+  if (emailIdentity) {
+    // ...
+  } else if (googleIdentity) {
+    // ...
+  }
+  // ...
+}
+```
+
+```ts title=src/server/tasks.ts
+import { findUserIdentity } from '@wasp/auth/user.js'
+
+export const createTask: CreateTask<...>  = async (args, context) => {
+  const emailIdentity = findUserIdentity(context.user, 'email')
+  const googleIdentity = findUserIdentity(context.user, 'google')
+  if (emailIdentity) {
+    // ...
+  } else if (googleIdentity) {
+    // ...
+  }
+  // ...
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Custom Login and Signup Actions
+
+Let's take a look how you can use the `Auth` and `AuthIdentity` entities to create custom login and signup actions. For example, you might want to create a custom signup action that creates a user in your app and also creates a user in a third-party service.
```

Comment:
Makes sense, I'll just link to the example we have for specific providers 👍

Reply:
I've linked to specific providers, but I've kept this example as well since it talks about using the `createUser` helper or just using Prisma directly if they want.